internal/util.go

```diff
@@ -20,9 +21,21 @@ func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]i
 	typeOf := reflect.TypeOf(s)
 	for i := 0; i < valueOf.NumField(); i++ {
 		field := typeOf.Field(i)
-		tagValue := field.Tag.Get("json")
-		if _, ok := extras[tagValue]; ok {
-			delete(extras, tagValue)
+
+		lowerField := strings.ToLower(field.Name)
+		if _, ok := extras[lowerField]; ok {
```

Comment:
This `if` condition (and the one below) aren't necessary: `delete` either succeeds or is a no-op.